test/service/test_manifest.py

```diff
@@ -755,6 +755,90 @@ def test_full_metadata(self):
             self.assertIn(freeze(expected_300_fastq2_row), rows)
             self.assertTrue(all(len(row) == len(expected_fastq1_row) for row in rows))
 
+    @mock_sts
+    @mock_s3
+    def test_full_metadata_missing_fields(self):
+        self.maxDiff = None
+        self._index_canned_bundle(("f79257a7-dfc6-46d6-ae00-ba4b25313c10", "2018-09-14T133314.453337Z"))
+        # moto will mock the requests.get call so we can't hit localhost; add_passthru let's us hit the server
+        # see this GitHub issue and comment: https://github.com/spulec/moto/issues/1026#issuecomment-380054270
+        with ResponsesHelper() as helper:
+            helper.add_passthru(self.base_url)
+            storage_service = StorageService()
+            storage_service.create_bucket()
+
+            params = {'filters': json.dumps({'project': {'is': ['Single of human pancreas']}}), 'format': 'full'}
+            response = self.get_manifest(params)
+            self.assertEqual(200, response.status_code, 'Unable to download manifest')
+            tsv_file1 = csv.reader(response.iter_lines(decode_unicode=True), delimiter='\t')
+            fieldnames1 = next(tsv_file1)
+
+            params = {'filters': json.dumps({'project': {'is': ['Mouse Melanoma']}}), 'format': 'full'}
+            response = self.get_manifest(params)
+            self.assertEqual(200, response.status_code, 'Unable to download manifest')
+            tsv_file2 = csv.reader(response.iter_lines(decode_unicode=True), delimiter='\t')
+            fieldnames2 = next(tsv_file2)
+
+            symmetric_diff = set(fieldnames1) ^ set(fieldnames2)
+
+            expected_diff = {
```

Comment:
This is a nifty test. Since there is already a test that asserts the presence of individual columns in the full TSV, could we change this to `assertNotEmpty(symmetric_diff)`? And maybe also add `assertNotEmpty(union)`?

Reply:
Added assertion of the sets intersection along with the symmetric diff. No `assertNotEmpty()` available so I used `assertGreater(len(diff), 0)` instead.